pkg/middlewares/loopstop/loop_stop_test.go

```diff
@@ -0,0 +1,30 @@
+package loopstop
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestServeHTTP(t *testing.T) {
+	req, err := http.NewRequest(http.MethodGet, "", nil)
+	require.NoError(t, err)
+
+	m, err := NewLoopStop("myRouter", http.HandlerFunc(func(_ http.ResponseWriter, _ *http.Request) {}))
+	require.NoError(t, err)
+
+	recorder := httptest.NewRecorder()
+	m.ServeHTTP(recorder, req)
+
+	assert.Equal(t, http.StatusOK, recorder.Code)
+
+	require.Equal(t, m.routerNameHash, req.Header.Get(xTraefikRouter))
```

Comment:
Don't you think you should test that `xTraefikRouter` is the hash of `myRouter`?
If for some reason `m.routerNameHash` is empty, the test will pass but the middleware is broken.